sympy/physics/continuum_mechanics/beam.py

```diff
@@ -396,6 +396,12 @@ def apply_load(self, value, start, order, end=None):
         start = sympify(start)
         order = sympify(order)
 
+        if start.is_real:
+            if start > self.length or (end != None and end > self.length):
```

Comment:
These make me nervous -- if symbolic or non-real values are sent then this test will fail for reasons other than intended.
```python
if start.is_real is False or end and end.is_real is False: raise TypeError
if Or(start > self.length,
   (self.length if end is None else end) > self.length): raise ValueError for out of bounds
```

Reply:
What happens if you pass `Symbol('x1', positive=True)` for start?
```python
>>> if Symbol('x1', positive=True)>1:print('yes')
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/base/data/home/apps/s~sympy-live-hrd/56.412575950366508090/sympy/sympy/core/relational.py", line 229, in __nonzero__
    raise TypeError("cannot determine truth value of Relational")
TypeError: cannot determine truth value of Relational
```

Reply:
What about `fuzzy_or` along with `Gt`, `Lt`?